src/Database/Relations/Concerns/DefinedConstraints.php

```diff
@@ -11,20 +14,18 @@ trait DefinedConstraints
     /**
      * Set the defined constraints on the relation query.
      *
+     * This method is kept for backwards compatibility, but is no longer being called directly by Storm when
+     * initializing relations. Constraints are now applied on an as-needed basis.
```

Comment:
This method used to pull double functionality. It set constraints on the relation object, and set constraints on the query, but this method called both on the relation object itself when retrieving the relation.

Setting constraints on the relation now is only needed when using property style relations, because in method style relations, you would add the constraints directly in the method, such as conditions, ordering, default models, timestamps, etc.

The query constraints are also only automatically applied on property style relations, and only when needed in the HasRelationships concern. I also saw it being used in the Relation Controller.

Reply:
Oh, and yes, we should mark it deprecated